server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala

```diff
@@ -232,15 +233,62 @@ private[deploy] class Worker(
       override def run(): Unit = Utils.tryLogNonFatalError {
         unavailablePeers.entrySet().asScala.foreach(entry => {
           if (System.currentTimeMillis() - entry.getValue >
-            replicateFastfailDuration) {
+            REPLICATE_FAST_FAIL_DURATION) {
             unavailablePeers.remove(entry.getKey)
           }
         })
       }
-    }, 0, replicateFastfailDuration, TimeUnit.MILLISECONDS)
+    }, 0, REPLICATE_FAST_FAIL_DURATION, TimeUnit.MILLISECONDS)
+
+    if (RssConf.metricsSystemEnable(conf)) {
+      logInfo(s"Metrics system enabled!")
+      metricsSystem.start()
+
+      var port = RssConf.workerPrometheusMetricPort(conf)
+      var initialized = false
+      while (!initialized) {
+        try {
+          val httpServer = new HttpServer(
+            new HttpServerInitializer(
+              new HttpRequestHandler(metricsSystem.getPrometheusHandler)), port)
+          httpServer.start()
+          initialized = true
+        } catch {
+          case e: Exception =>
+            logWarning(s"HttpServer pushPort $port may already exist, try pushPort ${port + 1}.", e)
+            port += 1
+            Thread.sleep(1000)
+        }
+      }
+    }
+
+    cleaner = new Thread("Cleaner") {
+      override def run(): Unit = {
+        while (true) {
+          val expiredShuffleKeys = cleanTaskQueue.take()
+          try {
+            cleanup(expiredShuffleKeys)
+          } catch {
+            case e: Throwable =>
+              logError("Cleanup failed", e)
+          }
+        }
+      }
+    }
+
+    pushDataHandler.init(this)
+    replicateHandler.init(this)
+    fetchHandler.init(this)
```

Comment:
How about just
```
class ChunkFetchHandler(
  val conf: TransportConf, 
   worker: Worker) extends BaseHandler with Logging {
  var streamManager = new OneForOneStreamManager()
  var source: WorkerSource = worker.xxx
  .......

  override def registered =. worker.registered
```

Reply:
> I've considered this but I don't think it's a good idea for the handler to own a worker instance.

Maybe we can define registered as a function, then we don't need to do `init()` after registered.

Reply:
> Maybe we can define registered as a function, then we don't need to do init() after registered.

Call init() inside Worker's init() is because fields are guaranteed to be initialized there.